### PR TITLE
feat(vrl): add `uuid_v7` function

### PR DESCRIPTION
**Pull request by ppaulweber**
This PR introduces the following changes:

* add remap behavioral test for VRL `uuid_v7()` function
* provided cue website documentation for `uuid_v7()` function
* depends on vectordotdev/vrl/pull/738

### CODE REVIEW

**Review by brett0000FF — APPROVED** [review]

File: website/cue/reference/remap/functions/uuid_v7.cue [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package metadata
+
+remap: functions: uuid_v7: {
+	category:    "Random"
+	description: """
+		Generates a random [UUIDv7](\(urls.uuidv7)) string.
```
Comment (pront) [inline website/cue/reference/remap/functions/uuid_v7.cue:6]: Is this the `uuidv7` url defined?

**Review by pront — APPROVED** [review]

**Comment by ppaulweber** [comment]
FYI @pront and @brett0000FF: I've updated quickly the `cue` file to include the optional parameter for the `uuid_v7()` function [1] and provided more usage examples [2].

[1] https://github.com/vectordotdev/vector/pull/20048/files#diff-a86a5523ed4e7e8309ed7311fe5580c3b2fbe0caa28a416182c01bc6b90fe9e3R9-R17

[2] https://github.com/vectordotdev/vector/pull/20048/files#diff-a86a5523ed4e7e8309ed7311fe5580c3b2fbe0caa28a416182c01bc6b90fe9e3R22-R42

File: website/cue/reference/remap/functions/uuid_v7.cue (hunk 1 above)
Comment (pront) [reply website/cue/reference/remap/functions/uuid_v7.cue:6]: Most urls are defined in `website/cue/reference/urls.cue`.

Forgot to mention, you can render these locally with `cd website && make run-production-site-locally`.

Comment (ppaulweber) [reply website/cue/reference/remap/functions/uuid_v7.cue:6]: @pront thx, I've added the missing URL now, see:
https://github.com/vectordotdev/vector/pull/20048/files#diff-d93ed3f0b2ace4e29cca5ab64a8493f2691d3f3351ae7624794a84fd9ab8bdc3R530

Sadly the `cd website && make run-production-site-locally` fails for me locally with the following error message which looks like the upstream is currently broken in that regard:

```
Error: error building site: process: readAndProcessContent: "/vector/website/content/en/docs/reference/configuration/sinks/aws_cloudwatch_metrics.md:1:1": unknown kind "sink" in front matter
```

Comment (jszwedko) [reply website/cue/reference/remap/functions/uuid_v7.cue:6]: Interesting, I see that error too. I'll look into that.

Comment (jszwedko) [reply website/cue/reference/remap/functions/uuid_v7.cue:6]: Fixed here: https://github.com/vectordotdev/vector/pull/20058

Comment (ppaulweber) [reply website/cue/reference/remap/functions/uuid_v7.cue:6]: @jszwedko thx, already rebased and works now ... looks as expected in the newly rendered function documentation:

image

**Review by jszwedko — APPROVED** [review]
Awesome, thanks @ppaulweber ! We can merge this after the next VRL update.

**Comment by ppaulweber** [comment]
@jszwedko great to hear :tada: I just provided a rebase to the lastest upstream.

**Comment by jszwedko** [comment]
@ppaulweber there appears to be a formatting CI failure here. Do you mind taking a look? I think you just need to run `cut fmt`

**Comment by jszwedko** [comment]
Also maybe merge in `master` to pull in the VRL update.

**Comment by ppaulweber** [comment]
@jszwedko just updated the PR to the latest upstream -- the `cue fmt` had no effect. 

If it fails again, I'll investigate the issue in more depth :detective:

**Comment by jszwedko** [comment]
> @jszwedko just updated the PR to the latest upstream -- the `cue fmt` had no effect.
> 
> If it fails again, I'll investigate the issue in more depth 🕵️

Ah, you'll need to provide the file path like `cue fmt <path/to/file.cue>`. I was unclear on that front.

**Comment by ppaulweber** [comment]
@jszwedko thx, I've formatted all the CUE files I've touched and now it looks much better. I've ran the `make check-docs` locally, see:

```
[vector]$ make check-docs
cargo vdev check docs
Skipping cue files format validation - reserved for CI
Validating cue files correctness...
Success! The contents of the sources in the "./website/cue" directory are valid
```

**Comment by jszwedko** [comment]
Perfect, thanks!

**Pull request merged**